docs/src/use/migrate-to-9.0.0.md

```diff
@@ -308,6 +309,21 @@ export default [
 
 **Related issue(s):** [#17488](https://github.com/eslint/eslint/issues/17488)
 
+## <a name="no-inner-declarations"></a> `no-inner-declarations` has a new default behavior with a new option
+
+ESLint v9.0.0 introduces a new option in `no-inner-declarations` rule called `blockScopeFunctions` which by default allows block-level `functions` in strict mode when `languageOptions.ecmaVersion` is set to `2015` or above.
+
+```js
+/*eslint no-inner-declarations: "error"*/
+"use strict";
+
+if (test) {
+    function foo () { }  // no error
+}
+```
+
```

Comment:
Can you add a "to address" paragraph like the other things in the migration guide? It should explain how to retain the old behavior of the rule.

Reply:
Done!